packages/loopring_v3/contracts/test/TestMigrationBridgeConnector.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "./TestSwapper.sol";
+import "../core/iface/IExchangeV3.sol";
+import "../lib/AddressUtil.sol";
+import "../lib/ERC20.sol";
+import "../lib/ERC20SafeTransfer.sol";
+import "../lib/MathUint.sol";
+import "../thirdparty/SafeCast.sol";
+import "../aux/bridge/IBridge.sol";
+
+
+/// Migrates from Loopring to ... Loopring!
+/// @author Brecht Devos - <brecht@loopring.org>
+contract TestMigrationBridgeConnector is IBridgeConnector
+{
+    using AddressUtil       for address payable;
+    using ERC20SafeTransfer for address;
+    using MathUint          for uint;
+    using SafeCast          for uint;
+
+    struct GroupSettings
+    {
+        address token;
+    }
+
+    struct UserSettings
+    {
+        address to;
+    }
+
+    IExchangeV3        public immutable exchange;
+    IDepositContract   public immutable depositContract;
+
+    IBridge            public immutable bridge;
+
+    constructor(
+        IExchangeV3 _exchange,
+        IBridge     _bridge
+        )
+    {
+        exchange = _exchange;
+        depositContract = _exchange.getDepositContract();
+
+        bridge = _bridge;
+    }
+
+    function processCalls(ConnectorGroup[] memory groups)
```

Comment:
A lot of code in this method are the same as in TestSwappperBridgeConnector's `processCalls`. Shall we extract the same logic into Bridge.sol?

Reply:
Both are just test implementations and actual implementations may need to do very different things depending on the dapp that needs to be interacted with. So I don't think we can put more into Bridge without losing flexibility in the connectors to do what is needed. If some functionality can be reused an extra external library seems better.